services/frontend-service/pkg/handler/releasetrain.go

```diff
@@ -33,6 +36,26 @@ func (s Server) handleReleaseTrain(w http.ResponseWriter, req *http.Request, env
 		return
 	}
 
+	if s.AzureAuth {
+		signature := req.Header.Get("signature")
+		if signature == "" {
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte("Missing signature header"))
+			return
+		} else {
```

Comment:
```suggestion
		}
```
you don't need an "else" after a return. This helps reduce the indentation